pkg/controller/snapshotter/snapshotter.go

```diff
@@ -160,94 +165,87 @@ func (vs *volumeSnapshotter) getSnapshotDataFromSnapshotName(snapshotName string
 	// fails the snapshot creation.
 	for {
-		// TODO(xyang): Use status returned from DescribeSnapshot to update snapshot
-		conditions, completed, err = plugin.DescribeSnapshot(&snapshotDataObj)
-		if !completed {
-			glog.Infof("Snapshot %s creation is not complete yet. Status: [%#v] Retrying...", snapshotName, conditions)
-			if err != nil {
-				// Don't bail out here. The error message could be "current snapshot status is: creating".
-				glog.Infof("Message received when creating snapshot %s: %v", snapshotName, err)
+		conditions, _, err = (*plugin).DescribeSnapshot(&snapshotDataObj)
+		newstatus = vs.getSimplifiedSnapshotStatus(conditions)
+		if newstatus == statusPending {
+			if newstatus != status {
+				status = newstatus
+				glog.Infof("Snapshot %s creation is not complete yet. Status: [%#v] Retrying...", snapshotName, conditions)
+				if err != nil {
+					// Don't bail out here. The error message could be "current snapshot status is: creating".
+					glog.Infof("Message received when creating snapshot %s: %v", snapshotName, err)
+				}
+				// UpdateVolmeSnapshot status
+				_, err = vs.UpdateVolumeSnapshot(snapshotName, conditions)
+				if err != nil {
+					glog.Errorf("Error updating volume snapshot %s: %v", snapshotName, err)
+					// NOTE(xyang): Return error if failed to update VolumeSnapshot after
+					// create snapshot request is sent to the plugin and VolumeSnapshotData is updated
+					return fmt.Errorf("Failed to update VolumeSnapshot for snapshot %s: %v", snapshotName, err)
+				}
 			}
 			time.Sleep(createVolumeSnapshotDataInterval)
-		} else {
-			// TODO(xyang): Use status returned from DescribeSnapshot.
-			glog.Infof("waitForPluginSnapshot: Snapshot %s creation is complete: %#v", snapshotName, conditions)
-			break
-		}
-	}
-
-	if completed {
-		glog.Infof("Snapshot %s successfully created and it is ready.", snapshotName)
-		// UpdateVolmeSnapshot
-		snapConditions := []crdv1.VolumeSnapshotCondition{
-			{
-				Type:               crdv1.VolumeSnapshotConditionReady,
-				Status:             v1.ConditionTrue,
-				Message:            "Snapshot created succsessfully",
-				LastTransitionTime: metav1.Now(),
-			},
-		}
-		_, err = vs.UpdateVolumeSnapshot(snapshotName, &snapConditions, true)
-		if err != nil {
-			glog.Errorf("Error updating volume snapshot %s: %v", snapshotName, err)
-			// NOTE(xyang): Return error if failed to update VolumeSnapshot after
-			// create snapshot request is sent to the plugin and VolumeSnapshotData is updated
-			return fmt.Errorf("Failed to update VolumeSnapshot for snapshot %s: %v", snapshotName, err)
-		}
-
-		snapDataConditions := []crdv1.VolumeSnapshotDataCondition{
-			{
-				Type:               crdv1.VolumeSnapshotDataConditionReady,
-				Status:             v1.ConditionTrue,
-				Message:            "Snapshot data created succsessfully",
-				LastTransitionTime: metav1.Now(),
-			},
+			continue
```

Comment:
I think it will be nice to have exponential backoff instead of a fixed internal. https://github.com/jpillora/backoff
But we can add it later.

Reply:
Sure, I'll take a look.  It's probably better to add this after the basic workflow change is done.